### ISSUE TITLE
Feature request: intensity

### ISSUE THREAD

**Issue by DoctorJellyface** [jonls/redshift#249]
I have the Twilight app on my phone, which is similar to Redshift, and by default it comes with the night temperature at 1500K but only 30% intensity, so it's more red but less dark. I could really use this on Redshift. Thanks!

**Comment by toolforger** [comment]
As http://jonls.dk/redshift/ , using Redshift for intensity will not work very well because it manipulates just the gamma ramp, not the backlight. You'd get a kind of whiteout with 30% intensity.

**Comment by DoctorJellyface** [comment]
No, not the backlight, I mean the intensity of the "redness".

**Comment by DoctorJellyface** [comment]
So would this be possible?

**Comment by toolforger** [comment]
Ah sorry, somehow I missed your answer.

1,500K redness means going by 4,000K from daylight color (6,500K). 30% of that would be a change of just 1,200K, so you'd end at 5,300K.
Is that what you want?

**Comment by DoctorJellyface** [comment]
Well, at 4500K my computer screen is more yellow than red. On 1500K it's the colour I want, but it's way too intensive. What I want is to turn the intensity down, so I still have a red screen, just with not as much red, as in twilight.

**Comment by toolforger** [comment]
Sorry, but I simply do not understand what you want. The only parameters that I know that can be changed are color shift and intensity, but you say both are fine but the overall result is still not what you want - but I don't know what you want changed.
I don't have Twilight so I can't look by myself how the effect should look like.

**Comment by DoctorJellyface** [comment]
Sorry, I'll try to explain it better.

Imagine you have two layers, the bottom layer is the screen and the top layer is Redshift.  The screen would be normal but the redshift layer has a transparent red over it, making the screen look red too.

In this scenario the intensity would be the transparency of the red layer.

Btw, that's basically how it's done on Android.

**Comment by toolforger** [comment]
Well, Redshift isn't adding a layer of transparent red, it's translating colors.
Would there be any advantage in adding a transparent layer?

**Comment by DoctorJellyface** [comment]
I don't think so, except, maybe Wayland support. Would it be possible to achieve a similar effect just by translating colours?

**Comment by toolforger** [comment]
I didn't mean technical advantages, I meant visual advantages.
"Similar" is in the eye of the beholder; unless you define the effect, there's no saying what needs to be done. (This is about the third time I'm trying to get a description of the visual effect out of you. Unless you can provide one, there is very little hope anything can be done for you.)

**Comment by DoctorJellyface** [comment]
Hang with me here, I'm trying my best ;) I did actually record a quick video of the function with my mums phone, quality's not great, but it shows what I mean. I'll upload it as soon as I have time.

**Comment by DoctorJellyface** [comment]
OK, **finally** got to it. [Here](https://www.dropbox.com/s/emk0vt4n9zcjhcz/twilight-intensity.webm?dl=0)'s the wonderful video. Be warned that I was doing two things at once while recording, so it looks jumpy, the focus is sometimes weird, but you get the point from it. Plus, I learned how to convert videos from the CLI! So at least something positive.

**Comment by toolforger** [comment]
Can you upload to somewhere accessible without Javascript?
(I consider JS too much of a security risk to allow anybody into my system, sorry.)

**Comment by DoctorJellyface** [comment]
Sure, just change the `dl=0` to `dl=1` in the URL, like [so](https://www.dropbox.com/s/emk0vt4n9zcjhcz/twilight-intensity.webm?dl=1).

**Comment by toolforger** [comment]
Okay, got around to looking at it.
Indeed looks like Twilight is adding a layer of semitransparent Red, with "intensity" essentially translating to the alpha value.

Sure that looks different - is it better? Obviously you think it is :-) - however, it's not obvious to me in what ways it is better; it may well be that you think it's better just because you're used to it.
One way to decide that might be if you installed Redshift instead of Twilight and let a week or two go for getting used to it, then check its effects: Quality of sleep, ease of reading and identifying symbols on the display, i.e. does it work better for you? Worse? Different but meh?

**Comment by DoctorJellyface** [comment]
Thanks for your time. I agree, it's mostly just different. The legibility isn't affected by neither. I don't know about the sleep, I take that mostly as a side effect. I redden my screen because I just can't stand looking at a blazing white screen in the evening, it tears my eyes apart. I prefer the transparent way solely for the way it affects colours. I don't mind watching movies/videos with twilight, but I turn off redshift because it just changes the way the film looks too much.

But anyway, that's just my point of view. If you manage to implement this I'll be very thankful, if notn nevermind.

**Comment by DoctorJellyface** [comment]
Sorry, I'm on mobile and the keyboard is just making fun of me. It's the second time I pressed something on the page accidentally.

**Comment by jonls** [comment]
No worries @DoctorJellyface Thanks for reporting this issues, I'll keep your suggestions in mind.
